test/unit/response.test.js

```diff
@@ -121,4 +124,43 @@ describe('Response', function () {
             expect(response.size().body).to.eql(20);
         });
     });
+
+    ((typeof window === 'undefined') ? describe : describe.skip)('fromModule helper', function () {
+        var response;
+
+        before(function (done) {
+            request.get('https://echo.getpostman.com/get', function (err, res) {
+                if (err) {
+                    return done(err);
+                }
+
+                response = new Response(Response.fromModule(res)).toJSON();
+                done();
+            });
+        });
+
+        it('should correctly return the processed response', function () {
```

Comment:
EXTENSIVE tests has to be written using different forms of response bodies, headers (duplicate (array-style) headers), and all subsequent levels of objects needs to be verified.